frontend/src/tests/instructor-permission-test.js

```diff
@@ -516,51 +469,254 @@ export function instructorsPermissionTests(api) {
 
     it("fetch applications", async () => {
         await switchToInstructorOnlyUser();
-        let resp = await apiGET(
+        const resp = await apiGET(
             `/instructor/sessions/${session.id}/applications`
         );
         expect(resp).toHaveStatus("success");
 
         await restoreDefaultUser();
     });
 
-    it("fetch Ddahs", async () => {
-        // If a user is not in Instructors table, it is not considered an instructor
-        // for the purpose of fetching DDAHs - so we create one
-        const newInstructor = {
-            first_name: "Jane",
-            last_name: "Smith",
-            email: "jane.smith@gmail.com",
-            utorid: instructorUser.utorid,
-        };
-        let resp = await apiPOST(`/admin/instructors`, newInstructor);
-        expect(resp).toHaveStatus("success");
+    describe("Ddah permissions", () => {
+        let position;
+        let instructor;
+        let assignment;
+        let foreignAssignment;
+
+        beforeAll(async () => {
+            await restoreDefaultUser();
+
+            // Get our instructor
+            let resp = await apiGET(`/admin/instructors`);
+            expect(resp).toHaveStatus("success");
+            instructor = resp.payload.find(
+                (instructor) => instructor.utorid === instructorUser.utorid
+            );
+            expect(instructor).toBeDefined();
+
+            // Fetch the position related to our instructor
+            resp = await apiGET(`/admin/sessions/${session.id}/positions`);
+            position = resp.payload.find((position) =>
+                position.instructor_ids.find((id) => id === instructor.id)
+            );
+            expect(position).toBeDefined();
+
+            // Create another assignment (with no DDAH)
+            // for position associated with our instructor
+            const newAssignment = {
+                note: "",
+                applicant_id: databaseSeeder.seededData.applicant.id,
+                position_id: position.id,
+                start_date: "2019-09-02T00:00:00.000Z",
+                end_date: "2019-12-31T00:00:00.000Z",
+            };
+            resp = await apiPOST("/admin/assignments", newAssignment);
+            expect(resp).toHaveStatus("success");
+            assignment = resp.payload;
```

Comment:
Maybe you should add a check if the assignment is defined since you are assuming that the assignment is defined in the future. It is fine without it but you cannot really over-test it :)